### PR TITLE
Fix CI failure with SWIG::Python on Ubuntu 24.04

### LINKED ISSUE #988: CI failure w/ SWIG::Python on Ubuntu 24.04
### Bug description

As of today, the CI fails while Swig compiles icub.i for python:

```console
2024-10-14T06:09:47.4155353Z /home/runner/work/icub-main/icub-main/install/include/yarp/sig/Image.h:42: Error: Type error. Expecting an integral type
...
2024-10-14T06:09:47.4197493Z /home/runner/work/icub-main/icub-main/install/include/yarp/sig/Image.h:68: Error: Type error. Expecting an integral type
2024-10-14T06:09:47.4870606Z /home/runner/work/icub-main/icub-main/install/include/yarp/dev/IAxisInfo.h:24: Error: Type error. Expecting an integral type
2024-10-14T06:09:47.4873013Z /home/runner/work/icub-main/icub-main/install/include/yarp/dev/IAxisInfo.h:25: Error: Type error. Expecting an integral type
2024-10-14T06:09:47.4875108Z /home/runner/work/icub-main/icub-main/install/include/yarp/dev/IAxisInfo.h:27: Error: Type error. Expecting an integral type
2024-10-14T06:09:47.5572523Z /home/runner/work/icub-main/icub-main/install/include/yarp/dev/IInteractionMode.h:18: Error: Type error. Expecting an integral type
2024-10-14T06:09:47.5574884Z /home/runner/work/icub-main/icub-main/install/include/yarp/dev/IInteractionMode.h:19: Error: Type error. Expecting an integral type
2024-10-14T06:09:47.5577180Z /home/runner/work/icub-main/icub-main/install/include/yarp/dev/IInteractionMode.h:21: Error: Type error. Expecting an integral type
```

It seems that there are problems dealing with `yarp::conf::createVocab32()`.

cc @traversaro 

### Steps to reproduce

Link to failed CI:
- https://github.com/robotology/icub-main/actions/runs/11321719276/job/31481204884

### Expected behavior

_No response_

### Example repository

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by pattacini**
Fixes #988

Update CI configuration to use Ubuntu 24.04 and custom SWIG 4.2.1

* Change the CI configuration in `.github/workflows/ci.yml` to use `ubuntu-24.04` instead of `ubuntu-22.04`.
* Add steps to install a custom build of SWIG 4.2.1 for Ubuntu 24.04.
* Remove the step to install SWIG via `sudo apt install`.
* Ensure the custom SWIG 4.2.1 is used for the build process on Ubuntu 24.04.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/robotology/icub-main/pull/999?shareId=a324a996-0248-44c6-9f6f-307b95f6081f).

### CODE REVIEW

**Comment by pattacini** [comment]
The relevant CI is:
- https://github.com/robotology/icub-main/actions/runs/12239795583

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -56,12 +56,23 @@ jobs:
...
+    - name: Install Custom SWIG 4.2.1 [Ubuntu]
+      if: matrix.os == 'ubuntu-24.04'
+      shell: bash
+      run: |
+        mkdir -p ${GITHUB_WORKSPACE}/swig
+        cd ${GITHUB_WORKSPACE}/swig
+        wget https://github.com/robotology/robotology-vcpkg-ports/releases/download/storage/swig_4_2_1_ubuntu_24_04.zip
+        unzip swig_4_2_1_ubuntu_24_04.zip
+        export PATH=${GITHUB_WORKSPACE}/swig/bin:$PATH
+        export SWIG_LIB=${GITHUB_WORKSPACE}/swig/share/swig/4.2.1
```
Comment (pattacini) [inline .github/workflows/ci.yml:75]: Hi @traversaro 

The CI is still failing.

Do you if I have to set up the CMake params to build YARP as done in https://github.com/robotology/robotology-superbuild/blob/master/cmake/RobotologySuperbuildLogic.cmake#L60-L61 or setting the env vars `PATH` and `SWIG_LIB` as done here should be sufficient?

Comment (pattacini) [reply .github/workflows/ci.yml:75]: I realized now that the zip contains a further level of path.

Comment (pattacini) [reply .github/workflows/ci.yml]: We've got still v4.2.0 installed somewhere.

image

Comment (pattacini) [reply .github/workflows/ci.yml]: It's in the runner image:
- https://github.com/actions/runner-images/issues/1203

Comment (traversaro) [reply .github/workflows/ci.yml]: The `export` command is not persistent across jobs steps. To set the `PATH` you either need to do  this in the step in which you actually use SWIG (like in the cmake step) or by setting the PATH persistently via https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#adding-a-system-path and https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#example-of-writing-an-environment-variable-to-github_env .

Comment (pattacini) [reply .github/workflows/ci.yml]: Correct! I'll do it!

**Pull request merged**